modules/Classroom/components/MessageList/MessageList.tsx

```diff
@@ -1,15 +1,49 @@
+import { useEffect, useState } from "react";
+
 import { Box, Text } from "@mantine/core";
-import dayjs from "dayjs";
+import { io, Socket } from "socket.io-client";
 
+import { WS_BASE_URL } from "@/shared/constants/env.constants";
+import { useAppSelector } from "@/shared/redux/hooks";
+import { selectAuthenticatedUser } from "@/shared/redux/reducers/user.reducer";
 import { useGetClassroomMessagesQuery } from "@/shared/redux/rtk-apis/messages/messages.api";
+import { IMessage } from "@/shared/redux/rtk-apis/messages/messages.interface";
 
 import MessageCard from "../MessageCard/MessageCard";
 import { IMessageListProps } from "./MessageList.interface";
 import useStyles from "./MessageList.styles";
 
 const MessageList: React.FC<IMessageListProps> = ({ classroomId }) => {
-  const { data: messages } = useGetClassroomMessagesQuery(classroomId);
+  const { data: initialMessages, isSuccess, refetch } = useGetClassroomMessagesQuery(classroomId);
   const { classes } = useStyles();
+  const [messages, setMessages] = useState<IMessage[]>([]);
+  const user = useAppSelector(selectAuthenticatedUser);
+
+  useEffect(() => {
+    refetch();
+  }, [classroomId, refetch]);
```

Comment:
Why do we need to do this?

Reply:
It's because, when a user goes to different classroom(B) and then come back to the classroom(A) the messages should be updated, like in classroom(A) there might be other user's who is chatting.

Reply:
You can look into [refetchOnMountOrArgChange](https://redux-toolkit.js.org/rtk-query/usage/cache-behavior#encouraging-re-fetching-with-refetchonmountorargchange) rather than manually calling refetch